From 6486ac7e68ecb58e1dc85bdf2e1b7555b677606d Mon Sep 17 00:00:00 2001
From: Sam Leffler <sleffler@chromium.org>
Date: Tue, 19 Jul 2011 15:34:48 -0700
Subject: [PATCH] CHROMIUMOS: wpa_supplicant: do not expire bss entries when a
 scan is aborted

If a scan request completes prematurely do not run the normal bss expire
logic as it removes entries that have not been seen for N complete scans.
This should only slightly delay expiring old entries as the only aborted
scans are/will be background scans.

BUG=chromium-os:11485
TEST=manual:depends on other changes

Change-Id: I4dae4b7fa81f967d400d6630aa8df01f544fc37a
Reviewed-on: http://gerrit.chromium.org/gerrit/4347
Reviewed-by: Gary Morain <gmorain@chromium.org>
Reviewed-by: Paul Stewart <pstew@chromium.org>
Tested-by: Sam Leffler <sleffler@chromium.org>
Commit-Ready: Sam Leffler <sleffler@chromium.org>
---
 wpa_supplicant/bss.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/wpa_supplicant/bss.c b/wpa_supplicant/bss.c
index 7d01a5f..4994e6e 100644
--- a/wpa_supplicant/bss.c
+++ b/wpa_supplicant/bss.c
@@ -738,8 +738,9 @@ void wpa_bss_update_end(struct wpa_supplicant *wpa_s, struct scan_info *info,
 	struct wpa_bss *bss, *n;
 
 	os_get_reltime(&wpa_s->last_scan);
-	if (!new_scan)
-		return; /* do not expire entries without new scan */
+	/* only expire entries with a new complete scan */
+	if (!new_scan || (info != NULL && info->aborted))
+		return;
 
 	dl_list_for_each_safe(bss, n, &wpa_s->bss, struct wpa_bss, list) {
 		if (wpa_bss_in_use(wpa_s, bss))
-- 
2.2.0.rc0.207.ga3a616c

